### ISSUE TITLE
Updating the calendars leads to ForeignKeyViolation

### ISSUE THREAD

**Issue by Jorricks** [Jorricks/DutyBoard#6]
This happens when you are trying to rename or delete an calendar.
```
02-23 16:02 duty_board.worker_loop INFO     Updating plugin calendars in the database.
Traceback (most recent call last):
  File "/home/bigdata/venv/lib64/python3.9/site-packages/sqlalchemy/engine/base.py", line 1900, in _execute_context
    self.dialect.do_execute(
  File "/home/bigdata/venv/lib64/python3.9/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.ForeignKeyViolation: update or delete on table "calendar" violates foreign key constraint "on_call_event_calendar_uid_fkey" on table "on_call_event"
DETAIL:  Key (uid)=(CDI-IS) is still referenced from table "on_call_event".


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/bigdata/venv/bin/dutyBoard", line 8, in <module>
    sys.exit(cli())
  File "/home/bigdata/venv/lib64/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/bigdata/venv/lib64/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/bigdata/venv/lib64/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/bigdata/venv/lib64/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/bigdata/venv/lib64/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/bigdata/venv/lib64/python3.9/site-packages/duty_board/main.py", line 27, in worker
    worker_loop.enter_loop()
  File "/home/bigdata/venv/lib64/python3.9/site-packages/duty_board/worker_loop.py", line 133, in enter_loop
    queries.sync_duty_calendar_configurations_to_postgres(session, plugin.duty_calendar_configurations)
  File "/home/bigdata/venv/lib64/python3.9/site-packages/duty_board/alchemy/queries.py", line 155, in sync_duty_calendar_configurations_to_postgres
    session.query(Calendar).where(Calendar.uid.not_in(all_described_calendar_uids)).delete()
  File "/home/bigdata/venv/lib64/python3.9/site-packages/sqlalchemy/orm/query.py", line 3222, in delete
    result = self.session.execute(
  File "/home/bigdata/venv/lib64/python3.9/site-packages/sqlalchemy/orm/session.py", line 1714, in execute
    result = conn._execute_20(statement, params or {}, execution_options)
  File "/home/bigdata/venv/lib64/python3.9/site-packages/sqlalchemy/engine/base.py", line 1705, in _execute_20
    return meth(self, args_10style, kwargs_10style, execution_options)
  File "/home/bigdata/venv/lib64/python3.9/site-packages/sqlalchemy/sql/elements.py", line 334, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/home/bigdata/venv/lib64/python3.9/site-packages/sqlalchemy/engine/base.py", line 1572, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/bigdata/venv/lib64/python3.9/site-packages/sqlalchemy/engine/base.py", line 1943, in _execute_context
    self._handle_dbapi_exception(
  File "/home/bigdata/venv/lib64/python3.9/site-packages/sqlalchemy/engine/base.py", line 2124, in _handle_dbapi_exception
    util.raise_(
  File "/home/bigdata/venv/lib64/python3.9/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/bigdata/venv/lib64/python3.9/site-packages/sqlalchemy/engine/base.py", line 1900, in _execute_context
    self.dialect.do_execute(
  File "/home/bigdata/venv/lib64/python3.9/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) update or delete on table "calendar" violates foreign key constraint "on_call_event_calendar_uid_fkey" on table "on_call_event"
DETAIL:  Key (uid)=(CDI-IS) is still referenced from table "on_call_event".

[SQL: DELETE FROM calendar WHERE (calendar.uid NOT IN (%(uid_1_1)s, %(uid_1_2)s, %(uid_1_3)s, %(uid_1_4)s, %(uid_1_5)s, %(uid_1_6)s, %(uid_1_7)s, %(uid_1_8)s, %(uid_1_9)s, %(uid_1_10)s, %(uid_1_11)s, %(uid_1_12)s, %(uid_1_13)s, %(uid_1_14)s, %(uid_1_15)s, %(uid_1_16)s, %(uid_1_17)s, %(uid_1_18)s, %(uid_1_19)s, %(uid_1_20)s, %(uid_1_21)s, %(uid_1_22)s, %(uid_1_23)s, %(uid_1_24)s, %(uid_1_25)s, %(uid_1_26)s, %(uid_1_27)s, %(uid_1_28)s, %(uid_1_29)s))]
[parameters: {'uid_1_1': 'PQAAAAA',....,'uid_1_29': 'PZAAAAA'}]
(Background on this error at: https://sqlalche.me/e/14/gkpj)
```

**Comment by Jorricks** [comment]
Quick and dirty solution. Wipe the database and start over.

**Comment by Jorricks** [comment]
Should be fixed in DutyBoard 0.3.0
